src/System.Private.CoreLib/shared/System/IO/Path.cs

```diff
@@ -401,7 +388,7 @@ public static string Combine(params string[] paths)
                 }
                 else
                 {
-                    char ch = builder[builder.Length - 1];
+                    char ch = builder[^1];
```

Comment:
Please revert all use of ^1 in this file.  This file ends up getting compiled into some downlevel assemblies that do not have access to Index.